src/Thunks/WS2_32.hpp

```diff
@@ -1463,6 +1463,34 @@ namespace YY::Thunks
     }
 #endif
 
+#if (YY_Thunks_Target < __WindowsNT6_1_SP1)
+#ifndef WSA_FLAG_NO_HANDLE_INHERIT
+#define WSA_FLAG_NO_HANDLE_INHERIT    0x80
+#endif
+
+    __DEFINE_THUNK(
+    ws2_32,
+    24,
+    SOCKET,
+    WSAAPI,
+    WSASocketW,
+        _In_ int af,
+        _In_ int type,
+        _In_ int protocol,
+        _In_opt_ LPWSAPROTOCOL_INFOW lpProtocolInfo,
+        _In_ GROUP g,
+        _In_ DWORD dwFlags
+        )
+    {
+        if (auto const pWSASocketW = try_get_WSASocketW())
+        {
+            // This flag is supported on Windows 7 with SP1, Windows Server 2008 R2 with SP1, and later
+            dwFlags &= ~WSA_FLAG_NO_HANDLE_INHERIT;
```

Comment:
You should check the Windows version for that logic.

Reply:
> We should not do that for all cases because YY-Thunks is not the Rust-specific project..

But it is an unsupported flag anyway if you are not running Windows 7 SP1

Reply:
> > We should not do that for all cases because YY-Thunks is not the Rust-specific project..
> 
> But it is an unsupported flag anyway if you are not running Windows 7 SP1

You should check the Windows version dynamically first. You should not include that logic for OSes newer than Windows 7 Service Pack 1.

Reply:
> > > We should not do that for all cases because YY-Thunks is not the Rust-specific project..
> > 
> > 
> > But it is an unsupported flag anyway if you are not running Windows 7 SP1
> 
> You should check the Windows version dynamically first. You should not include that logic for OSes newer than Windows 7 Service Pack 1.

Oh isn't that controlled by the version check macro?

Reply:
> > > > We should not do that for all cases because YY-Thunks is not the Rust-specific project..
> > > 
> > > 
> > > But it is an unsupported flag anyway if you are not running Windows 7 SP1
> > 
> > 
> > You should check the Windows version dynamically first. You should not include that logic for OSes newer than Windows 7 Service Pack 1.
> 
> Oh isn't that controlled by the version check macro?

The version check macro is compile time check for people choose the different base OS level.

Reply:
> > > > > We should not do that for all cases because YY-Thunks is not the Rust-specific project..
> > > > 
> > > > 
> > > > But it is an unsupported flag anyway if you are not running Windows 7 SP1
> > > 
> > > 
> > > You should check the Windows version dynamically first. You should not include that logic for OSes newer than Windows 7 Service Pack 1.
> > 
> > 
> > Oh isn't that controlled by the version check macro?
> 
> The version check macro is compile time check for people choose the different base OS level.

Right. Is using GetVersionW enough to check or there is better way?

Reply:
> > > > > > We should not do that for all cases because YY-Thunks is not the Rust-specific project..
> > > > > 
> > > > > 
> > > > > But it is an unsupported flag anyway if you are not running Windows 7 SP1
> > > > 
> > > > 
> > > > You should check the Windows version dynamically first. You should not include that logic for OSes newer than Windows 7 Service Pack 1.
> > > 
> > > 
> > > Oh isn't that controlled by the version check macro?
> > 
> > 
> > The version check macro is compile time check for people choose the different base OS level.
> 
> Right. Is using GetVersionW enough to check or there is better way?

https://github.com/Chuyu-Team/YY-Thunks/blob/841f26f632021f887783855d41f6573a8c401a19/src/Thunks/api-ms-win-core-string.hpp#L297

You should use the YY-Thunks's version check API.

Reply:
Okay but what about Windows 7 SP1 specifically?

Reply:
> Okay but what about Windows 7 SP1 specifically?

It looks you are using PEB, it's OK. But I hope @mingkuang-Chuyu can expose the extended version check function which can check the build number.